plugins/currency/bitcoin/bitcoind_

```diff
@@ -134,20 +134,36 @@ def main():
         print(error, file=sys.stderr)
         return False
 
+    if request_var == 'balance':
+        # we use getbalance*s* (plural) as old betbalance is being deprecated,
+        # and we have to calculate total balance (owned and watch-only) manually now.
+        (info, error) = bitcoin.getbalances()
+
+        total = 0
+
+        for wallet_mode in ('mine', 'watchonly'):
+            if wallet_mode in info:
+                total += info[wallet_mode]['trusted']
+
```

Comment:
The five lines above could be beautifully combined into a list comprehension:
```python
total = sum(info[wallet_mode]['trusted']
            for wallet_mode in ('mine', 'watchonly')
            if wallet_mode in info)
```

Sorry in advance for lecturing - I just like python's comprehensions a lot ...

(of course, it is just a matter of taste)

Reply:
I like using algorithms too in C++ ;) . Fixed.